### PR TITLE
plugin(userVoiceShow): Add voice activity status icons in Member and DM list

### PR DESCRIPTION
**Pull request by Johannes7k75**
Adds Activity Icons from the [BD plugin](https://betterdiscord.app/plugin/VoiceActivity) which was created by [Neodymium](https://betterdiscord.app/developer/Neodymium)

### CODE REVIEW

**Comment by TheKodeToad** [comment]
Surely it wouldn't be too hard to make this without copying parts of it?

**Comment by Johannes7k75** [comment]
My question is maybe dumb, but why? His licence states that the code can be reused.

**Comment by Methosss** [comment]
speaker icon next to the nickname with hover to see where he is would be great

**Comment by TheKodeToad** [comment]
> My question is maybe dumb, but why? His licence states that the code can be reused.

That doesn't mean you should just take it (I didn't look too closely though)

**Comment by Vendicated** [comment]
> That doesn't mean you should just take it (I didn't look too closely though)

Why not?

**Comment by Johannes7k75** [comment]
> speaker icon next to the nickname with hover to see where he is would be great

Where should the icon be? (Maybe a screenshot?)

**Comment by TheKodeToad** [comment]
> > That doesn't mean you should just take it (I didn't look too closely though)
> 
> Why not?

i'm a bit late lol
imo copying something isn't much fun - and if it's done without permission or credit in the ui the developer probably won't like it

**Comment by Marocco2** [comment]
> > > That doesn't mean you should just take it (I didn't look too closely though)
> > 
> > 
> > Why not?
> 
> i'm a bit late lol imo copying something isn't much fun - and if it's done without permission or credit in the ui the developer probably won't like it
It's MIT licensed and he already posted the license on the first commit in PR, so it's good (he doesn't need futher permissions because they are already granted by this very license)

**Comment by slikie** [comment]
> Where should the icon be? (Maybe a screenshot?)

How about something like this? plugin from https://betterdiscord.app/plugin/VoiceActivity
image

**Comment by Johannes7k75** [comment]
The icons are already there
image

**Comment by slikie** [comment]
Maybe the issue title need to be more clear.

**Comment by slikie** [comment]
Actually I get a crash with this build - 

```
77cadfdbb8448a8c8452.js:3489 TypeError: Cannot destructure property 'user' of 'undefined' as it is undefined.
    at Object.patchDmList (index.tsx:122:21)
 ---
    patchDmList: ({ user }: UserProps) => {
        if (!settings.store.showVoiceActivityIcons) return null;

        return (
            <ErrorBoundary noop >
                <div className={vaCl("iconContainer")}>
                    <VoiceActivityIcon user={user} />
                </div>
            </ErrorBoundary >
        );
    },
```

**Comment by Johannes7k75** [comment]
I now have fixed the MemberList.
But the Dmlist is not fixed yet.

**Comment by Johannes7k75** [comment]
Now MemberList and DmList work

**Comment by slikie** [comment]
@Vendicated is there any issue before merging this?

**Comment by Johannes7k75** [comment]
Fixed last the last discord update.

File: src/plugins/userVoiceShow/index.tsx [hunk 1]
```diff
@@ -93,22 +108,49 @@ export default definePlugin({
...
+            </ErrorBoundary>
+
+        );
+    },
+
     patches: [
-        // above message box
         {
-            find: ".lastEditedByContainer",
+            find: ".showCopiableUsername",
```
Comment (AutumnVN) [inline src/plugins/userVoiceShow/index.tsx:124]:
```suggestion
            find: ".lastEditedByContainer,",
```

File: src/plugins/userVoiceShow/index.tsx [hunk 2]
```diff
@@ -93,22 +108,49 @@ export default definePlugin({
...
             replacement: {
-                match: /\(0,\i\.jsx\)\(\i\.\i,{user:\i,setNote/,
+                match: /\(0,\w\.jsx\)\(\w{2},{user:\w,setNote/,
```
Comment (AutumnVN) [inline src/plugins/userVoiceShow/index.tsx:126]:
```suggestion
                match: /\(0,\i\.jsx\)\(\i\.\i,{user:\i,setNote/,
```

File: src/plugins/userVoiceShow/index.tsx [hunk 3]
```diff
@@ -93,22 +108,49 @@ export default definePlugin({
...
+                // paste my fancy custom button above the message field
                 replace: "$self.patchPopout(arguments[0]),$&",
             }
         },
-        // below username
         {
             find: ".USER_PROFILE_MODAL",
             replacement: {
-                match: /\.body.+?displayProfile:\i}\),/,
+                match: /\(\)\.body.+?displayProfile:\i}\),/,
```
Comment (AutumnVN) [inline src/plugins/userVoiceShow/index.tsx:134]:
```suggestion
                match: /\.body.+?displayProfile:\i}\),/,
```

File: src/plugins/userVoiceShow/index.tsx [hunk 4]
```diff
@@ -93,22 +108,49 @@ export default definePlugin({
...
+                // paste my fancy custom button below the username
                 replace: "$&$self.patchModal(arguments[0]),",
             }
+        },
+        {
+            // Patch Member List
+            find: "default.MEMBER_LIST_ITEM_AVATAR_DECORATION_PADDING",
```
Comment (AutumnVN) [inline src/plugins/userVoiceShow/index.tsx:141]:
```suggestion
            find: ".MEMBER_LIST_ITEM_AVATAR_DECORATION_PADDING)",
```

File: src/plugins/userVoiceShow/index.tsx [hunk 5]
```diff
@@ -93,22 +108,49 @@ export default definePlugin({
...
+            replacement: {
+                match: /avatar:\(\(/,
+                replace: "children:[$self.patchUserList(arguments[0], false)],$&",
+            }
+        },
+        {
+            // Patch Dm List
+            find: "PrivateChannel.renderAvatar",
+            replacement: {
+                match: /highlighted:.+name:.+decorators.+\}\)\}\),/,
```
Comment (AutumnVN) [inline src/plugins/userVoiceShow/index.tsx:151]:
```suggestion
                match: /highlighted:.+?name:.+?decorators.+?\}\)\}\),/,
```

**Comment by D3SOX** [comment]
I still use this PR I think it's good, are you going to reopen it?

**Comment by Johannes7k75** [comment]
Oh, i just thought i had changed my branch on my repo. 
Sure i reopen it

File: src/plugins/userVoiceShow/index.tsx [hunk 6]
```diff
@@ -93,22 +108,48 @@ export default definePlugin({
...
                 match: /\.body.+?displayProfile:\i}\),/,
+                // paste my fancy custom button below the username
                 replace: "$&$self.patchModal(arguments[0]),",
             }
+        },
+        {
+            // Patch Member List
+            find: ".MEMBER_LIST_ITEM_AVATAR_DECORATION_PADDING)",
+            replacement: {
+                match: /avatar:\(\(/,
```
Comment (D3SOX) [inline src/plugins/userVoiceShow/index.tsx:142]:
```suggestion
                match: /avatar:\i\(\i,\i\)/,
```

This broke I fixed it this way

Comment (Johannes7k75) [reply src/plugins/userVoiceShow/index.tsx]: I forgot to push a fix i did two week ago.
i changed it to 
```js
match: /avatar:(\i){1,2}/
```

**Comment by Nuckyz** [comment]
Superseded by https://github.com/Vendicated/Vencord/commit/bae9f8cbe696b3f4ddfd41f80e0bfdb5ccc3a755, thanks regardless!

**Pull request closed without merging**